### PR TITLE
namespace project

### PR DESCRIPTION
**Pull request by cchris28**
shows how this can be added. Persuant discussion we probably do not want to merge in current form but I put it up for review and comment.

I see I accidentally included a change that addresses https://github.com/lincc-frameworks/python-project-template/issues/193 . I removed that change (you can see both versions of the change in the first two commits)

### CODE REVIEW

File: copier.yml [hunk 1]
```diff
@@ -113,4 +113,4 @@ _skip_if_exists:
 
 # Tasks to execute after generating or updating a project.
 _tasks:
-    - "git init --initial-branch=main"
+    - "git init . && { git branch -m master main 2>/dev/null || true; }"
```
Comment (cchris28) [inline copier.yml]: this should be removed from this PR
Even so it does not work on git 1.8 which is the (ancient) version available on epyc.
I do have a solution for that although I think it rather ugly.

File: copier.yml [hunk 2]
```diff
@@ -113,4 +113,4 @@ _skip_if_exists:
 
 # Tasks to execute after generating or updating a project.
 _tasks:
-    - "git init . && { git branch -m master main 2>/dev/null || true; }"
+    - "git init . && echo 'ref: refs/heads/main' >.git/HEAD"
```
Comment (cchris28) [inline copier.yml:126]: again, this should be removed from this pr

**Comment by drewoldag** [comment]
There is some discussion happening about creating a RAIL-specific version of the Python project template that would be used for the (rather large amount) of RAIL package restructuring and renaming that is going to happen in the next few weeks. 

RAIL is already configured to use the RAIL namespace. So I wonder if it would be possible to port this work over to the new RAIL-specific fork of the python project template.

One thing that Eric Charles noted when he was experimenting with PPT was that the _version file needed to be moved. He called that out in issue #205

**Comment by drewoldag** [comment]
All in all, though, this looks like a reasonable unit of work, and would show how to modify a package to be used as a namespace.

**Comment by cchris28** [comment]
The only thing I did with the version was move it to the namespace package. This implementation renames src to <namespace> instead of pushing the packages down. If Eric moved the version file to be alongside the __init__.py file that marks the subpackage then it is equivalent I think.

I think it will easy to adapt this to the rail specific version. Slightly more work if rail prefers the src/rail/<subpackage> incarnation over the src -> rail/<subpackage> version. But not much.

**Comment by drewoldag** [comment]
Probably not going to get to this in the near future. Going to close out this PR. We can always reopen at a later date.

**Pull request closed without merging**